etcdserver/storage.go

```diff
@@ -109,14 +108,6 @@ func readWAL(waldir string, snap walpb.Snapshot) (w *wal.WAL, id, cid types.ID,
 // It must ensure that, after upgrading, the most recent version is present.
 func upgradeDataDir(baseDataDir string, name string, ver version.DataDirVersion) error {
 	switch ver {
-	case version.DataDir0_4:
-		plog.Infof("converting v0.4 log to v2.0")
-		err := migrate.Migrate4To2(baseDataDir, name)
-		if err != nil {
-			plog.Fatalf("failed to migrate data-dir (%v)", err)
-			return err
-		}
-		fallthrough
```

Comment:
I do not think so. We have to stop worrying about the 0.4 user at some point. And 2.2 is a great time for that in my opinion. We have supporting 0.4 migration for two versions (2.0, 2.1). And 2.1 is a barrier, we tell users explicitly they should not upgrade from 0.4 to 2.1. But yea, we still support that in case some one does it wrong.